unknown warning group '-Wunused-local-typedefs'

[Travis CI log](https://api.travis-ci.org/jobs/124514361/log.txt?deansi=true) shows
```
$ g++ --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 6.0 (clang-600.0.54) (based on LLVM 3.5svn)
Target: x86_64-apple-darwin13.4.0
Thread model: posix
```
with the following warning, but it suggests the actual compiler used is GCC. The Travis docs mention `apple-gcc42` being installed in the [OS X environment](https://docs.travis-ci.com/user/osx-ci-environment/#Compilers-and-Build-toolchain).
```
/Users/travis/build/BioMedIA/MIRTK/Modules/PointSet/src/PointLocator.cc:59:36: warning: unknown warning group '-Wunused-local-typedefs', ignored [-Wunknown-pragmas]
    #pragma GCC diagnostic ignored "-Wunused-local-typedefs"
```

--- a/Applications/src/evaluate-surface-overlap.cc
+++ b/Applications/src/evaluate-surface-overlap.cc
@@ -53,15 +53,15 @@
     #pragma warning(disable: 4291) // no matching operator delete found
     #pragma warning(disable: 4996) // _CRT_SECURE_NO_WARNINGS
   // Disable "Unused typedef 'ElementType'" in flann/ground_truth.h
+  #elif defined(__clang__) // *also* defines __GNUG__!
+    #pragma clang diagnostic push
+    #if !defined(__has_warning) || __has_warning("-Wunused-local-typedefs")
+      #pragma clang diagnostic ignored "-Wunused-local-typedefs"
+    #endif
   #elif defined(__GNUG__)
     #pragma GCC diagnostic push
     #if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 7)
       #pragma GCC diagnostic ignored "-Wunused-local-typedefs"
-    #endif
-  #elif defined __clang__
-    #pragma clang diagnostic push
-    #if __clang_major__ > 3 || (__clang_major == 3 && __clang_minor > 5)
-      #pragma clang diagnostic ignored "-Wunused-local-typedefs"
     #endif
   #endif
   // Include FLANN
@@ -69,10 +69,10 @@
   // Enable warnings again
   #if defined(_MSC_VER)
     #pragma warning(pop)
+  #elif defined(__clang__)
+    #pragma clang diagnostic pop
   #elif defined(__GNUG__)
     #pragma GCC diagnostic pop
-  #elif defined __clang__
-    #pragma clang diagnostic pop
   #endif
 #endif // HAVE_FLANN
 

--- a/Applications/src/evaluate-surface-overlap.cc
+++ b/Applications/src/evaluate-surface-overlap.cc
@@ -55,10 +55,14 @@
   // Disable "Unused typedef 'ElementType'" in flann/ground_truth.h
   #elif defined(__GNUG__)
     #pragma GCC diagnostic push
-    #pragma GCC diagnostic ignored "-Wunused-local-typedefs"
+    #if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 7)
+      #pragma GCC diagnostic ignored "-Wunused-local-typedefs"
+    #endif
   #elif defined __clang__
     #pragma clang diagnostic push
-    #pragma clang diagnostic ignored "-Wunused-local-typedefs"
+    #if __clang_major__ > 3 || (__clang_major == 3 && __clang_minor > 5)
+      #pragma clang diagnostic ignored "-Wunused-local-typedefs"
+    #endif
   #endif
   // Include FLANN
   #include "flann/flann.hpp"

--- a/Modules/PointSet/src/PointLocator.cc
+++ b/Modules/PointSet/src/PointLocator.cc
@@ -54,15 +54,15 @@
     #pragma warning(disable: 4291) // no matching operator delete found
     #pragma warning(disable: 4996) // _CRT_SECURE_NO_WARNINGS
   // Disable "Unused typedef 'ElementType'" in flann/ground_truth.h
+  #elif defined(__clang__) // *also* defines __GNUG__!
+    #pragma clang diagnostic push
+    #if !defined(__has_warning) || __has_warning("-Wunused-local-typedefs")
+      #pragma clang diagnostic ignored "-Wunused-local-typedefs"
+    #endif
   #elif defined(__GNUG__)
     #pragma GCC diagnostic push
     #if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 7)
       #pragma GCC diagnostic ignored "-Wunused-local-typedefs"
-    #endif
-  #elif defined(__clang__)
-    #pragma clang diagnostic push
-    #if __clang_major__ > 3 || (__clang_major == 3 && __clang_minor > 5)
-      #pragma clang diagnostic ignored "-Wunused-local-typedefs"
     #endif
   #endif
   // Include FLANN
@@ -70,10 +70,10 @@
   // Enable warnings again
   #if defined(_MSC_VER)
     #pragma warning(pop)
-  #elif defined(__GNUG__)
-    #pragma GCC diagnostic pop
   #elif defined(__clang__)
     #pragma clang diagnostic pop
+  #elif defined(__GNUG__)
+    #pragma GCC diagnostic pop
   #endif
 #endif // HAVE_FLANN
 

--- a/Modules/PointSet/src/PointLocator.cc
+++ b/Modules/PointSet/src/PointLocator.cc
@@ -56,10 +56,14 @@
   // Disable "Unused typedef 'ElementType'" in flann/ground_truth.h
   #elif defined(__GNUG__)
     #pragma GCC diagnostic push
-    #pragma GCC diagnostic ignored "-Wunused-local-typedefs"
-  #elif defined __clang__
+    #if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 7)
+      #pragma GCC diagnostic ignored "-Wunused-local-typedefs"
+    #endif
+  #elif defined(__clang__)
     #pragma clang diagnostic push
-    #pragma clang diagnostic ignored "-Wunused-local-typedefs"
+    #if __clang_major__ > 3 || (__clang_major == 3 && __clang_minor > 5)
+      #pragma clang diagnostic ignored "-Wunused-local-typedefs"
+    #endif
   #endif
   // Include FLANN
   #include "flann/flann.hpp"
@@ -68,7 +72,7 @@
     #pragma warning(pop)
   #elif defined(__GNUG__)
     #pragma GCC diagnostic pop
-  #elif defined __clang__
+  #elif defined(__clang__)
     #pragma clang diagnostic pop
   #endif
 #endif // HAVE_FLANN

